.github/workflows/060_generate_benchmark_weights.yml

```diff
@@ -0,0 +1,73 @@
+name: Generate benchmark weights
+
+on:
+  workflow_dispatch:
```

Comment:
Is it possible to include a `pull_request` event that targets the development branch? This would ensure that the workflow runs automatically whenever a pull request is created or updated.

Additionally, we can utilize a path filter to execute the workflow only when changes affect the pallet files. Furthermore, we can narrow down the filter to include only `.rs` files in the `pallets` directory, thereby avoiding unnecessary runs when pallet `.md` files are updated.

Reply:
Good remark
You mean when we merge a pull request to development branch I guess?
Maybe it is too heavy... I would do it only when a release/tag is created what do you think?

But yes, in all cases it would make sense to add it with the filter you described, good suggestion.

Reply:
The workflow should ensure that we merge PRs containing updated weights.rs before approving and merging a PR, rather than after. also I prefer not to add automated commits against the main branch directly.

At first, I was thinking something like this is workable
```
on:
  pull_request:
    branches: development
```
`pull_request` event here will trigger the workflow run every time you push a commit to the feature branch but Limited to Pull Requests which target development branch. 

But now I’m rethinking that this is even more heavy. Unless the developer waits to open the PR until their branch is ready to merge, which we can’t enforce.
I don’t like either the idea of generating these files only on releases. I think it should be updated with any changes that affect the computation effort.

so, we need to run this on the feature branch only when we’re ready to merge, but it not as straightforward as it sounds. so here is what I can suggest:

- We can trigger the workflow with every push to the feature branch (see snippet above) but checking in the workflow for a specific keyword in the commit message e.g. [TO MERGE], if not exists we stop run immediately. it require to remember to adjust your last commit to work.   
- There are some workarounds online to avoid running a workflow when the PR is draft, so you can create it in Draft and push many commits as long as you want and when you turn it ready for review the CI runs.
- Your original approach to only run this manually is workable indeed, but then how we can enforce to not merge a PR without forgetting to run this workflow?
- I'm checking [slash-command](https://github.com/peter-evans/slash-command-dispatch) as well right now. it like dispatch it manually tbh but a lot nicer :)

Feel free to stick with manual dispatch for now, or to discuss any of these suggestions further.

Reply:
See new strategy here
https://github.com/threefoldtech/tfchain/issues/837#issuecomment-1935007481